Openvstorage/Framework documentation/structure/framework.md

```diff
@@ -0,0 +1,104 @@
+## Framework structure
+
+### Packaging
+Previously redhat (deprecated), now debian is used as default ubuntu.
+This module contains all installfiles that are needed for packaging and pre-, post- and regular installing ovs
+
+### Scripts
+Contains everything that is a `.sh` file or as script that needs to be run once.
+
+`/system/ovs.sh` contains the entrypoint to the ovs CLI in the Andes releases. This has been reworked though in develop and master, and this is now offloaded to the python click library.
+
+### Venv
+
+### OVS
+#### CLI 
+_MASTER ONLY_
+
+This folder contains all logic needed for the `click` library to build our refactored CLI.
+Every `ovs` subcommand has its subfolder, except for calls used once (-> `misc`)
+The calls are built dynamically starting from this folders' `__init__.py`. Check the `__init.py__`, `commands.py` and `entry.py` files for seeing its magic in action.
+Adding commands should be done accordingly: 
+- Add the module to the main `__init__.py` if necessary
+- implement call in `__init__.py` of module
+- add call to group
+
+#### Constants
+_MASTER ONLY_
+This module contains a multitude of strings of names, paths and other strings that are used throughout the codebase. Equivalent to regular i8n, its purpose is to prevent much refactoring in case some namestrings should change. And it is cleaner, in our opinion.
+
+Usage is pretty straightforward.
+
+
+#### DAL
+
+Ah, the DAL. A wonderful piece of black magic, obfuscating it design flaws through its obscurity.
+The DAL, in its essence, is a comfort (abstraction)layer on top of a db. 
+##### Hybrids
+Hybrids are name after their, well, _hybrid_ nature.
+They contain 
+- properties: static, fixed attributes that shouldn't be reloaded, or at least not often.
+- dynamics: dynamic, changing attributes that should reflect state or other frequently changing parameters of an object. 
+
+Now, get ready for some obscure magic.
+- relations: These relations reflect relations between 2 DAL objects that are a one-to-one or one-to-many relationship.
+
+But wait, there's more!
+
+- These relations only reflect one-to-one or one-to-many relationships. For many-to-many relationships, `junctions` were introduced. These files can be seen in the list of 
+the DAL hybrids as files named according to this format: `^j_.*`. 
+
+All these `properties`, `dynamics`, `relations` and `junctions` are manipulated into the DAL object upon creation with some metaprogramming in the `__init__` of the `DataObject` superclass.
+
+All these attributes can be accessed the same way however: 
+`DALObject.property` 
+
+
+#### Lists
+This is where the the comfortlayer on top of our sql DB becomes really handy. These lists contain logic to  gracefully implement SQL queries without having to worry about to much sql code.
```

Comment:
There is no SQL to be sighted within the Framework repository.
The DataList can take a query object definition which is dictated by our own code. The query object is then used to do client sided querying (massive drawback)!